python/ribasim/ribasim/model.py

```diff
@@ -376,6 +448,68 @@ def reset_contextvar(self) -> "Model":
         context_file_loading.set({})
         return self
 
+    def max_node_id(self) -> int:
+        return self.network.node.df.index.max()
+
+    def max_allocation_network_id(self) -> int:
+        m = self.network.node.df.allocation_network_id.max()
+        if pd.isna(m):
+            m = 0
+        return m
+
+    def merge_model(
```

Comment:
nit: Your function names are sometimes a bit verbose. It's not wrong, but you might want to think about it. This would result in `modela.merge_model(modelb)`. Purely `merge` would thus also work.